clicommand/env_set.go

```diff
@@ -0,0 +1,169 @@
+package clicommand
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/buildkite/agent/v3/env"
+	"github.com/buildkite/agent/v3/jobapi"
+	"github.com/urfave/cli"
+)
+
+const envSetHelpDescription = `Usage:
+  buildkite-agent env set [variable]
+
+Description:
+   Sets environment variable values in the current job execution environment. 
+   Existing variables will be overwritten.
+
+   Note that this subcommand is only available from within the job executor with
+   the ′job-api′ experiment enabled.
+   
+   Note that changes to the job environment variables only apply to subsequent
+   phases of the job. To read the new values of variables from within the
+   current phase, use ′env get′.
+
+   Note that Buildkite read-only variables cannot be overwritten.
+
+Example (sets the variables ′LLAMA′ and ′ALPACA′):
+
+    $ buildkite-agent env set LLAMA=Kuzco "ALPACA=Geronimo the Incredible"
+	Added:
+	+ LLAMA
+	Updated:
+	~ ALPACA	
+	
+Example (sets the variables ′LLAMA′ and ′ALPACA′ using a JSON object supplied
+over standard input):
+
+    $ echo '{"ALPACA":"Geronimo the Incredible","LLAMA":"Kuzco"}' | buildkite-agent env set --input-format=json --output-format=quiet -
+`
+
+type EnvSetConfig struct{}
+
+var EnvSetCommand = cli.Command{
+	Name:        "set",
+	Usage:       "Sets variables in the job execution environment",
+	Description: envSetHelpDescription,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:   "input-format",
+			Usage:  "Input format: plain or json",
+			EnvVar: "BUILDKITE_AGENT_ENV_SET_INPUT_FORMAT",
+			Value:  "plain",
+		},
+		cli.StringFlag{
+			Name:   "output-format",
+			Usage:  "Output format: quiet (no output), plain, json, or json-pretty",
+			EnvVar: "BUILDKITE_AGENT_ENV_SET_OUTPUT_FORMAT",
+			Value:  "plain",
+		},
+	},
+	Action: envSetAction,
+}
+
+func envSetAction(c *cli.Context) error {
+	cli, err := jobapi.NewDefaultClient()
```

Comment:
See the comments in `env_get.go` about `cli`.

Reply:
Done.